### PR TITLE
[feature] check redis auth failed

### PR DESCRIPTION
**Pull request by ftwbzhao**
Fix #3742 
@narfbg Thanks

### CODE REVIEW

File: system/libraries/Cache/drivers/Cache_redis.php [hunk 1]
```diff
@@ -302,7 +302,11 @@ protected function _setup_redis()
 
 		if (isset($config['password']))
 		{
-			$this->_redis->auth($config['password']);
+			if ( ! $this->_redis->auth($config['password']))
+			{
+			    log_message('debug', 'Cache: Redis authentication failed. Check the config.');
```
Comment (narfbg) [inline system/libraries/Cache/drivers/Cache_redis.php]: No 'check the config' please ...

**Comment by narfbg** [comment]
Still missing a changelog entry and still using spaces on lines 307, 308.

Please don't close the PR to submit a new one ... you can just commit & push your changes to the same branch and this one will be updated.

**Comment by ftwbzhao** [comment]
Sorry, I'm a newbie.

**Comment by narfbg** [comment]
Changelog entry still missing ...

**Comment by ftwbzhao** [comment]
Please check changelog entry too, Thanks :rose:

**Comment by narfbg** [comment]
You've added the changelog entry as a new feature under the 3.0.0 section.

It's a bug fix, and it has to be under 3.0.1.

**Comment by ftwbzhao** [comment]
Got It

**Comment by ftwbzhao** [comment]
Whoops, The Travis CI build failed.
But everything is ok.

---
 I am counting on you.

**Comment by narfbg** [comment]
You are still adding it under 3.0.0 ...

**Comment by ftwbzhao** [comment]
Do not want to be the first.. 
But give me 5 minutes（I'm sure this is the last commit 　:dash:

**Comment by narfbg** [comment]
You're not the first, and that's not a reason to produce improper documentation in the first place.

**Comment by ftwbzhao** [comment]
```
+Bug fixes for 3.0.1
+------------------
+
+-  Fixed a bug (#3744) - Redis :doc:`Caching <libraries/caching>` driver didn't handle authentication failures properly.
+
 Bug fixes for 3.0
 ------------------
```

Here is ok?

**Comment by narfbg** [comment]
No, it's not ok ... the "Bug fixes for 3.0.1" already exists in the changelog.

**Comment by ftwbzhao** [comment]
I tried my best.

**Pull request merged**